### ISSUE TITLE
Error compiling with --prod: Module build failed: TypeError: Cannot read property 'kind' of undefined

### ISSUE THREAD

**Issue by homer-jay** [wbhob/ionic-long-press#23]
```
ionic-app-script task: "build" 
[09:43:01]  Error: ./node_modules/ionic-long-press/fesm5/ionic-long-press.js Module build failed: TypeError: Cannot read 
            property 'kind' of undefined at isAngularDecoratorMetadataExpression 
            (/home/user/myproject/node_modules/@angular-devkit/build-optimizer/src/transforms/scrub-file.js:245:35)
            at checkNodeForDecorators 
            (/home/user/myproject/node_modules/@angular-devkit/build-optimizer/src/transforms/scrub-file.js:67:21)
            at visitNodes 
            (/home/user/myproject/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:12699:30)
            at Object.forEachChild 
            (/home/user/myproject/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:12875:24)
            at checkNodeForDecorators 
            (/home/user/myproject/node_modules/@angular-devkit/build-optimizer/src/transforms/scrub-file.js:58:31)
            at visitNode 
            (/home/user/myproject/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:12690:24)
            at Object.forEachChild 
            (/home/user/myproject/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:12782:21)
            at checkNodeForDecorators 
            (/home/user/myproject/node_modules/@angular-devkit/build-optimizer/src/transforms/scrub-file.js:58:31)
            at visitNode 
            (/home/user/myproject/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:12690:24)
            at Object.forEachChild 
            (/home/user/myproject/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:12828:24)
            @ ./src/app/app.module.ts 72:0-51 @ ./src/app/app.module.ngfactory.js @ ./src/app/main.ts 
Error: ./node_modules/ionic-long-press/fesm5/ionic-long-press.js
```

Output of ionic info:

```
Ionic:

   ionic (Ionic CLI)  : 4.3.1 (/usr/local/lib/node_modules/ionic)
   Ionic Framework    : ionic-angular 3.9.2
   @ionic/app-scripts : 3.1.8

Cordova:

   cordova (Cordova CLI) : 8.1.2 (cordova-lib@8.1.1)
   Cordova Platforms     : android 7.1.4
   Cordova Plugins       : cordova-plugin-ionic-webview 1.2.1, (and 27 other plugins)

System:

   Android SDK Tools : 26.1.1 (/home/user/Android/Sdk)
   NodeJS            : v9.4.0 (/usr/local/bin/node)
   npm               : 5.10.0
   OS                : Linux 4.13
```

**Comment by homer-jay** [comment]
Both the current version (v2.0.2) and the previous one (v2.0.0) are NOT COMPATIBLE with Ionic 3. 
I've solved my compilation problem by downgrading `ionic-long-press` to version 1.5.0 and locking the dependency to exactly that version in the `package.json`. All the HammerJS configuration steps are no longer required with 1.5.0.

I've lost like 3 hours debugging this issue and was able to solve it with a version downgrade by mere chance. I was already considering removing the plugin from my project and would have had to code an alternative to the long press gesture. 

I'm leaving this issue open for just in case @wbhob is able to fix the `package.json` of the plugin so that the latest version requires at least Ionic 4. Ideally I'd have appreciated some kind of incompatibility error when I ran `npm install ionic-long-press` onto my Ionic 3 project. It would have been even better if npm had downloaded v1.5.0 automatically for me, but not sure if this is possible.

In case nothing can be done about this in the `package.json`, would it be possible to put a big warning message in the v2.0.2 documentation page?

**Comment by loicfavory** [comment]
I have the same issue with Ionic 4 when building it with --prod using:

Angular 8.2.14
Ionic 4.11.8

```
ERROR in ./node_modules/ionic-long-press/fesm2015/ionic-long-press.js
Module build failed (from ./node_modules/@angular-devkit/build-optimizer/src/build-optimizer/webpack-loader.js):
TypeError: Cannot read property 'kind' of undefined
    at isAngularDecoratorMetadataExpression (/home/loic/Documents/Clients/SITES/hellotribumobile/node_modules/@angular-devkit/build-optimizer/src/transforms/scrub-file.js:265:35)
    at checkNodeForDecorators (/home/loic/Documents/Clients/SITES/hellotribumobile/node_modules/@angular-devkit/build-optimizer/src/transforms/scrub-file.js:77:21)
    at visitNodes (/home/loic/Documents/Clients/SITES/hellotribumobile/node_modules/typescript/lib/typescript.js:16514:30)
    at Object.forEachChild (/home/loic/Documents/Clients/SITES/hellotribumobile/node_modules/typescript/lib/typescript.js:16740:24)
    at checkNodeForDecorators (/home/loic/Documents/Clients/SITES/hellotribumobile/node_modules/@angular-devkit/build-optimizer/src/transforms/scrub-file.js:68:31)
    at visitNode (/home/loic/Documents/Clients/SITES/hellotribumobile/node_modules/typescript/lib/typescript.js:16505:24)
    at Object.forEachChild (/home/loic/Documents/Clients/SITES/hellotribumobile/node_modules/typescript/lib/typescript.js:16635:21)
    at checkNodeForDecorators (/home/loic/Documents/Clients/SITES/hellotribumobile/node_modules/@angular-devkit/build-optimizer/src/transforms/scrub-file.js:68:31)
    at visitNode (/home/loic/Documents/Clients/SITES/hellotribumobile/node_modules/typescript/lib/typescript.js:16505:24)
    at Object.forEachChild (/home/loic/Documents/Clients/SITES/hellotribumobile/node_modules/typescript/lib/typescript.js:16703:24)
    at checkNodeForDecorators (/home/loic/Documents/Clients/SITES/hellotribumobile/node_modules/@angular-devkit/build-optimizer/src/transforms/scrub-file.js:68:31)
    at visitNode (/home/loic/Documents/Clients/SITES/hellotribumobile/node_modules/typescript/lib/typescript.js:16505:24)
    at Object.forEachChild (/home/loic/Documents/Clients/SITES/hellotribumobile/node_modules/typescript/lib/typescript.js:16692:24)
    at checkNodeForDecorators (/home/loic/Documents/Clients/SITES/hellotribumobile/node_modules/@angular-devkit/build-optimizer/src/transforms/scrub-file.js:68:31)
    at visitNode (/home/loic/Documents/Clients/SITES/hellotribumobile/node_modules/typescript/lib/typescript.js:16505:24)
    at Object.forEachChild (/home/loic/Documents/Clients/SITES/hellotribumobile/node_modules/typescript/lib/typescript.js:16599:21)
```

**Comment by wbhob** [comment]
have you installed hammerjs as a peer dependency?

**Comment by wbhob** [comment]
published v2.0.3 with peer dep @ionic/angular@>= 4

**Comment by alyeaaah** [comment]
I have the same issue with Ionic 4 & ionic-long-press 2.0.3

here is my ionic info output

```
Ionic:

   Ionic CLI                     : 6.10.0 (/usr/local/lib/node_modules/@ionic/cli)
   Ionic Framework               : @ionic/angular 4.7.1
   @angular-devkit/build-angular : 0.802.0
   @angular-devkit/schematics    : 8.2.0
   @angular/cli                  : 8.2.0
   @ionic/angular-toolkit        : 2.0.0

Cordova:

   Cordova CLI       : 9.0.0 (cordova-lib@9.0.1)
   Cordova Platforms : not available
   Cordova Plugins   : not available

Utility:

   cordova-res                          : 0.15.1
   native-run (update available: 1.0.0) : 0.2.9

System:

   Android SDK Tools : 26.1.1 (/Users/zaula/Documents/sdk)
   NodeJS            : v10.15.3 (/usr/local/bin/node)
   npm               : 6.13.6
   OS                : macOS Mojave
   Xcode             : Xcode 10.2.1 Build version 10E1001
```

this my package.json dependencies

```
  "dependencies": {
    "@angular/common": "^8.2.0",
    "@angular/core": "^8.2.0",
    "@ionic-native/core": "^5.11.0",
    "@ionic/angular": "^4.7.1",
    "@types/hammerjs": "^2.0.36",
    "cordova-android": "^8.1.0",
    "ionic-long-press": "2.0.3",
    "hammerjs": "^2.0.8",
    ...
}
```
and this is my error output
```
ERROR in ./node_modules/ionic-long-press/fesm2015/ionic-long-press.js
Module build failed (from ./node_modules/@angular-devkit/build-optimizer/src/build-optimizer/webpack-loader.js):
TypeError: Cannot read property 'kind' of undefined
    at isAngularDecoratorMetadataExpression (/Users/myUser/Documents/Files/PROJECT/myApp/node_modules/@angular-devkit/build-optimizer/src/transforms/scrub-file.js:265:35)
    at checkNodeForDecorators (/Users/myUser/Documents/Files/PROJECT/myApp/node_modules/@angular-devkit/build-optimizer/src/transforms/scrub-file.js:77:21)
    at visitNodes (/Users/myUser/Documents/Files/PROJECT/myApp/node_modules/typescript/lib/typescript.js:16631:30)
    at Object.forEachChild (/Users/myUser/Documents/Files/PROJECT/myApp/node_modules/typescript/lib/typescript.js:16859:24)
    at transformer (/Users/myUser/Documents/Files/PROJECT/myApp/node_modules/@angular-devkit/build-optimizer/src/transforms/scrub-file.js:63:16)
    at transformSourceFileOrBundle (/Users/myUser/Documents/Files/PROJECT/myApp/node_modules/typescript/lib/typescript.js:67384:57)
    at transformation (/Users/myUser/Documents/Files/PROJECT/myApp/node_modules/typescript/lib/typescript.js:83586:24)
    at transformRoot (/Users/myUser/Documents/Files/PROJECT/myApp/node_modules/typescript/lib/typescript.js:83606:82)
    at Object.map (/Users/myUser/Documents/Files/PROJECT/myApp/node_modules/typescript/lib/typescript.js:499:29)
    at Object.transformNodes (/Users/myUser/Documents/Files/PROJECT/myApp/node_modules/typescript/lib/typescript.js:83593:30)
    at emitJsFileOrBundle (/Users/myUser/Documents/Files/PROJECT/myApp/node_modules/typescript/lib/typescript.js:84089:32)
    at emitSourceFileOrBundle (/Users/myUser/Documents/Files/PROJECT/myApp/node_modules/typescript/lib/typescript.js:84043:13)
    at forEachEmittedFile (/Users/myUser/Documents/Files/PROJECT/myApp/node_modules/typescript/lib/typescript.js:83832:34)
    at Object.emitFiles (/Users/myUser/Documents/Files/PROJECT/myApp/node_modules/typescript/lib/typescript.js:84026:9)
    at emitWorker (/Users/myUser/Documents/Files/PROJECT/myApp/node_modules/typescript/lib/typescript.js:89951:33)
    at /Users/myUser/Documents/Files/PROJECT/myApp/node_modules/typescript/lib/typescript.js:89912:66
    at runWithCancellationToken (/Users/myUser/Documents/Files/PROJECT/myApp/node_modules/typescript/lib/typescript.js:90002:24)
    at Object.emit (/Users/myUser/Documents/Files/PROJECT/myApp/node_modules/typescript/lib/typescript.js:89912:20)
    at Object.transformJavascript (/Users/myUser/Documents/Files/PROJECT/myApp/node_modules/@angular-devkit/build-optimizer/src/helpers/transform-javascript.js:111:13)
    at Object.buildOptimizer (/Users/myUser/Documents/Files/PROJECT/myApp/node_modules/@angular-devkit/build-optimizer/src/build-optimizer/build-optimizer.js:119:35)
    at Object.buildOptimizerLoader (/Users/myUser/Documents/Files/PROJECT/myApp/node_modules/@angular-devkit/build-optimizer/src/build-optimizer/webpack-loader.js:36:40)
[ERROR] An error occurred while running subprocess ng.
```
it was fine on debug using `ionic cordova build android`, but this error show when I build in release mode with --prod `ionic cordova build --release android --prod`

any suggestion please?
cc @wbhob
